### PR TITLE
[stable/artifactory] Postgresql requirement / ingress support / nginx optional

### PR DESCRIPTION
**Pull request by joosthofman**
Removed Postgresql own deployment and created dependency to stable/Postgresql.
Updated artifactory version number to 5.8.0.
Updated notes and added ingress documentation.
Added Ingress support to artifactory.
Made the NGINX deployment optional by adding an enabled flag.
Fixed naming of components.

### CODE REVIEW

**Comment by joosthofman** [comment]
@jainishshah17 @eldada please review.

**Comment by eldada** [comment]
@joosthof - can you wait until we merge https://github.com/kubernetes/charts/pull/3200 ? It has a few changes that you'll need to merge with before applying your PR (which looks like a good path to take).

**Comment by jainishshah17** [comment]
@joosthof I agree with @eldada. Let's  wait till https://github.com/kubernetes/charts/pull/3200 gets merged.

**Comment by eldada** [comment]
@joosthof - https://github.com/kubernetes/charts/pull/3200 is merged. Please update your PR.

**Comment by joosthofman** [comment]
@eldada PR updated.

**Review by jainishshah17 — CHANGES_REQUESTED** [review]
Can you change chart version to 6.3.0 or 6.3.1

File: stable/artifactory/values.yaml
Comment (eldada) [inline stable/artifactory/values.yaml]: Since Artifactory is configure with persistence by default, I'd like the DB to also set like this.

File: stable/artifactory/Chart.yaml
Comment (eldada) [inline stable/artifactory/Chart.yaml]: Should start 6.3.0.

File: stable/artifactory/values.yaml
Comment (eldada) [inline stable/artifactory/values.yaml]: I think either nginx or ingress should be enabled. Both are set to `enabled: false`. Keeping with current chart, I suggest you leave the nginx `enabled: true`.

File: stable/artifactory/values.yaml
Comment (eldada) [inline stable/artifactory/values.yaml]: Artifactory in its nature is still a stateful app (although we hope to change this in the future). I think the default should stay `true` here.

File: stable/artifactory/README.md
Comment (eldada) [inline stable/artifactory/README.md:13]: Add the optional ingress.
Also, add a documentation section on the ingress and its supported parameters.
- What it does
- Using the TLS secret
- Whitelisting IPs

File: stable/artifactory/requirements.yaml
Comment (eldada) [inline stable/artifactory/requirements.yaml]: You can update to a later version

**Comment by joosthofman** [comment]
@jainishshah17 Fixed the chart version

**Review by eldada — CHANGES_REQUESTED** [review]

File: stable/artifactory/README.md
Comment (joosthofman) [reply stable/artifactory/README.md:13]: I added information about TLS secrets and how to use.
For the Ingress and whitelisting part I added a link to the kubernetes documentation.

**Review by eldada — APPROVED** [review]
Looks good.
/approve

**Comment by mattfarina** [comment]
@jainishshah17 @eldada [were you interested in setting up an OWNERS file for this chart?](https://codeengineered.com/blog/2017/kubernetes-chart-owners/)

**Comment by jainishshah17** [comment]
@mattfarina Yes!!

**Comment by mattfarina** [comment]
@jainishshah17 Then please submit a PR and go through the member process. You can use me as one of your references. The process for that is documented at https://github.com/kubernetes/community/blob/master/community-membership.md#member

File: stable/artifactory/Chart.yaml
Comment (mattfarina) [inline stable/artifactory/Chart.yaml]: Because the API to the chart change (e.g., values.yaml file structure) in a backwards breaking way this should be 7.0.0 instead of a minor bump. That is to follow [semver](http://semver.org) rules.

**Review by mattfarina — CHANGES_REQUESTED** [review]

**Review by mattfarina — APPROVED** [review]

**Pull request merged**
